Framework/DataObjects/inc/MantidDataObjects/ScanningWorkspaceBuilder.h

```diff
@@ -0,0 +1,112 @@
+#ifndef MANTID_DATAOBJECTS_SCANNINGWORKSPACEBUILDER_H_
+#define MANTID_DATAOBJECTS_SCANNINGWORKSPACEBUILDER_H_
+
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidDataObjects/DllConfig.h"
+#include "MantidIndexing/IndexInfo.h"
+#include "MantidKernel/DateAndTime.h"
+#include "MantidKernel/V3D.h"
+#include "MantidKernel/Quat.h"
+
+#include <vector>
+
+namespace Mantid {
+namespace DataObjects {
+
+/** ScanningWorkspaceBuilder : This is a helper class to make it easy to build a
+  scanning workspace (a workspace with moving detectors), where all the
+  information about the scan is known in advance. The constructor takes the
+  arguments for the basic construction, then checks are made for consistency as
+  other information about the scanning workspace is set.
+
+  Things that must be set for successful building:
+   - Number of detectors, number of time indexes and number of bins (set via the
+  constructor)
+   - The instrument set via setInstrument
+   - The time ranges set via setTimeRanges
+
+  Some helper methods exist for specific cases, such as the whole instrument
+  rotating around the sample.
+
+  One current limitation to note here, that is not a general restriction within
+  Mantid, is that every detector must have the same set of time indexes.
+
+  Copyright &copy; 2017 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+  National Laboratory & European Spallation Source
+
+  This file is part of Mantid.
+
+  Mantid is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 3 of the License, or
+  (at your option) any later version.
+
+  Mantid is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  File change history is stored at: <https://github.com/mantidproject/mantid>
+  Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class MANTID_DATAOBJECTS_DLL ScanningWorkspaceBuilder {
+public:
+  enum class IndexingType { DEFAULT, TIME_ORIENTED, DETECTOR_ORIENTED };
```

Comment:
Here `DEFAULT` means that `buildDefaultSpectrumDefinitions();` is called in `MatrixWorkspace::setIndexInfo()`, and no spectrum definitions are supplied.

I discussed this a little with @SimonHeybrock a while back, and it was decided not to promise any particular method for building the default spectrum definitions.

I wonder if my choice of names could be a little better here though...